session/features.go

```diff
@@ -0,0 +1,64 @@
+package session
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/qlik-oss/gopherciser/action"
+	"github.com/qlik-oss/gopherciser/logger"
+)
+
+type (
+	// Features maps feature flags and values
+	Features struct {
+		m  map[string]bool
+		mu sync.Mutex
+	}
+)
+
+// UpdateFeatureMap request features from server and updates feature map
+func (features *Features) UpdateFeatureMap(rest *RestHandler, host string, actionState *action.State, logEntry *logger.LogEntry) {
+	features.mu.Lock()
+	defer features.mu.Unlock()
+
+	req, err := rest.GetSync(fmt.Sprintf("%s/api/v1/features", host), actionState, logEntry, nil)
+	if err != nil {
+		return
+	}
+
+	if features.m == nil {
+		features.m = make(map[string]bool)
+	}
+
+	if err := jsonit.Unmarshal(req.ResponseBody, &features.m); err != nil {
+		actionState.AddErrors(errors.WithStack(err))
+		return
+	}
+}
+
+// IsFeatureEnabled check if feature flag is enabled
+func (features *Features) IsFeatureEnabled(feature string) (bool, error) {
```

Comment:
We could return two different error types here to be able to recover in different ways, if feature map is not fetched or feature does not exist.

Do not know if it will be needed though ^^

Reply:
yea probably won't be used, but easy to add so can do that